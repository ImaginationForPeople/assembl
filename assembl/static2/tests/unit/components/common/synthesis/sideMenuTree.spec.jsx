import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import SideMenuTree from '../../../../../js/app/components/synthesis/sideMenuTree';
import '../../../../helpers/setupTranslations';

describe('SideMenuTree component', () => {
  const indexGenerator = jest.fn(() => {});
  const commonProps = {
    rootIdea: {
      ancestors: [],
      id: '2222222',
      img: {
        externalUrl: 'http://foo.com/bar'
      },
      numContributors: 10,
      numPosts: 100,
      phaseIdentifier: 'phaseId',
      posts: {
        edges: []
      },
      synthesisTitle: 'Intersting idea',
      title: 'Cool story'
    },
    synthesisPostId: '12345',
    subIdeas: [
      {
        ancestors: [],
        id: '2222222',
        img: {
          externalUrl: 'http://foo.com/bar'
        },
        numContributors: 10,
        numPosts: 100,
        phaseIdentifier: 'phaseId',
        posts: {
          edges: []
        },
        synthesisTitle: 'Intersting idea',
        title: 'Cool story'
      },
      {
        ancestors: [],
        id: '33333',
        img: {
          externalUrl: 'http://foo.com/bar'
        },
        numContributors: 5,
        numPosts: 50,
        phaseIdentifier: 'phaseId',
        posts: {
          edges: []
        },
        synthesisTitle: 'Okay idea',
        title: 'Very nice story'
      }
    ],
    index: 1,
    slug: 'felixdebate',
    indexGenerator: indexGenerator
  };
  it('should render a 1 level SideMenuTree component', () => {
    const props = {
      ...commonProps,
      parents: []
    };
    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<SideMenuTree {...props} />);
    const result = shallowRenderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
  it('should render a 2 level SideMenuTree', () => {
    const props = {
      ...commonProps,
      parents: [1]
    };
    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<SideMenuTree {...props} />);
    const result = shallowRenderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});