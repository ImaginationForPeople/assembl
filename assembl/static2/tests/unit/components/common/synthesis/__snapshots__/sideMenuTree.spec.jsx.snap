// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenuTree component should render a 1 level SideMenuTree component 1`] = `
<div>
  <div
    className="link-container"
  >
    <Link
      className="side-menu-link-1"
      onClick={[Function]}
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/felixdebate/syntheses/12345/#2222222"
    >
       
      Cool story
    </Link>
    <span
      className="assembl-icon-down-open pointer"
      onClick={[Function]}
    />
  </div>
  <div />
</div>
`;

exports[`SideMenuTree component should render a 2 level SideMenuTree 1`] = `
<div>
  <div
    className="link-container"
  >
    <Link
      className="side-menu-link-2"
      onClick={[Function]}
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/felixdebate/syntheses/12345/#2222222"
    >
       
      Cool story
    </Link>
  </div>
  <div>
    <SideMenuTree
      ideaOnScroll={undefined}
      index={1}
      indexGenerator={[Function]}
      parents={
        Array [
          1,
          1,
        ]
      }
      rootIdea={
        Object {
          "ancestors": Array [],
          "id": "2222222",
          "img": Object {
            "externalUrl": "http://foo.com/bar",
          },
          "numContributors": 10,
          "numPosts": 100,
          "phaseIdentifier": "phaseId",
          "posts": Object {
            "edges": Array [],
          },
          "synthesisTitle": "Intersting idea",
          "title": "Cool story",
        }
      }
      slug="felixdebate"
      subIdeas={Array []}
      synthesisPostId="12345"
      url="/felixdebate/syntheses/12345/#2222222"
    />
    <SideMenuTree
      ideaOnScroll={undefined}
      index={2}
      indexGenerator={[Function]}
      parents={
        Array [
          1,
          1,
        ]
      }
      rootIdea={
        Object {
          "ancestors": Array [],
          "id": "33333",
          "img": Object {
            "externalUrl": "http://foo.com/bar",
          },
          "numContributors": 5,
          "numPosts": 50,
          "phaseIdentifier": "phaseId",
          "posts": Object {
            "edges": Array [],
          },
          "synthesisTitle": "Okay idea",
          "title": "Very nice story",
        }
      }
      slug="felixdebate"
      subIdeas={Array []}
      synthesisPostId="12345"
      url="/felixdebate/syntheses/12345/#33333"
    />
  </div>
</div>
`;
